src/NuGet.Clients/NuGet.VisualStudio.Implementation/Extensibility/VsFrameworkCompatibility.cs

```diff
@@ -0,0 +1,73 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.Linq;
+using System.Runtime.Versioning;
+using NuGet.Frameworks;
+using NuGet.VisualStudio.Implementation.Resources;
+
+namespace NuGet.VisualStudio
+{
+    [Export(typeof(IVsFrameworkCompatibility))]
+    public class VsFrameworkCompatibility : IVsFrameworkCompatibility
+    {
+        private readonly IFrameworkNameProvider _frameworkNameProvider;
+        private readonly IFrameworkCompatibilityListProvider _compatibilityListProvider;
+
+        public VsFrameworkCompatibility()
+        {
+            _frameworkNameProvider = DefaultFrameworkNameProvider.Instance;
+            _compatibilityListProvider = DefaultCompatibilityListProvider.Instance;
+        }
+
+        public IEnumerable<FrameworkName> GetNetStandardFrameworks()
+        {
+            return _frameworkNameProvider
+                .GetNetStandardVersions()
+                .OrderBy(f => f.Version)
+                .Select(GetFrameworkName);
```

Comment:
I would add .ToList() or .ToArray() here to enumerate the list. A 3rd party caller may end up holding onto internal nuget resources otherwise without realizing it.